bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/overlay/SearchHistoryMenu.java

```diff
@@ -92,7 +93,35 @@ public Control createContents(Composite parent) {
 		return table;
 	}
 
+	private void moveSelectionInTable(int indexShift) {
+		selectedIndexInTable += indexShift;
+		if (selectedIndexInTable < 0) {
+			selectedIndexInTable = table.getItemCount() - 1;
+		} else if (selectedIndexInTable > table.getItemCount() - 1) {
+			selectedIndexInTable = 0;
+		}
+		table.setSelection(selectedIndexInTable);
+		historyEntrySelectedCallback.accept(table.getSelection()[0].getText());
+	}
+
 	private void attachTableListeners() {
+		table.addListener(SWT.MouseMove, event -> {
+			Point point = new Point(event.x, event.y);
+			TableItem item = table.getItem(point);
+			if (item != null) {
+				table.setSelection(item);
+				selectedIndexInTable = table.getSelectionIndex();
+			}
+		});
+		table.addKeyListener(KeyListener.keyPressedAdapter(e -> {
```

Comment:
nitpicky: The same method already registered another KeyListener at the table. Would it make sense to combine them? And maybe switch over the `e.keyCode` in case of having multiple if/else clauses.

Reply:
Yes, it makes sense to combine them, good catch!

https://github.com/eclipse-platform/eclipse.platform.ui/commit/b6186f57fbec28d23260fc3a2f3d1101b910c3c5